### PR TITLE
v.rast.move: Move points by values from raster map

### PR DESCRIPTION
**Pull request by wenzeslaus**
Move of vector features positions by X and Y values given in two raster maps. This can move linear features without breaking the visual connection as points falling into the same raster cell are shifted by the same value.

The name is v.rast.move because it moves vector using raster values (operates on vectors but the big deal is that it is using raster to do the moving). Alternative name would be _v.rast.displacement_. Original name was _v.move.points_.

The one example included now shows the principle, moves in X direction both ways depending on the raster values. My use case is randomizing a set of lines (connected and not connected).

Unlike _v.perturb_ which moves points randomly, this works on lines and keeps the values at a given cell. Unlike _v.transform_ combined with raster values in attribute columns, this operates on individual points in the line, not on the whole line.

Copies first category, assumes layer 1. Drops attributes. 

Handles null values and extent mismatch. Warns about the nulls by default. Warns when integers are used (different null check is needed but not implemented).

### CODE REVIEW

**Comment by wenzeslaus** [comment]
I renamed the module from _v.move.points_ to _v.rast.move_ because it moves vector using raster values (operates on vectors but the big deal is that it is using raster to do the moving).

Tests are included, use pytest, and run locally. Example in the documentation is included, image is generated by a notebook. Pylint mostly passes locally.

File: src/vector/v.rast.move/v.rast.move.py [hunk 1]
```diff
@@ -0,0 +1,192 @@
...
+
+import grass.script as gs
+
+
+class NullValue(RuntimeError):
+    """Raised when null value is not allowed and encountered"""
+
+
+class OutOfBounds(RuntimeError):
+    """Raised when vector extent is large than region (raster) extent"""
```
Comment (petrasovaa) [inline src/vector/v.rast.move/v.rast.move.py]:
```suggestion
    """Raised when vector extent is larger than region (raster) extent"""
```

**Review by petrasovaa — CHANGES_REQUESTED** [review]
Looks like it works on lines only? What happens with areas or points? Consider documenting and adding a check.

File: src/vector/v.rast.move/v.rast.move.py (hunk 1 above)
Comment (wenzeslaus) [reply src/vector/v.rast.move/v.rast.move.py]: Thanks. Fixed.

**Comment by wenzeslaus** [comment]
> ...lines only? ...

Right, it works only for lines. Other types would be a reasonable extension. Points are easy to do without this tool.

I added a check which ignore other types and explained the behavior in the documentation. I also added a test which tests for this behavior.

**Review by petrasovaa — APPROVED** [review]

**Pull request merged**